index.js

```diff
@@ -713,6 +713,25 @@
       [$.Function, a, b, c, d],
       lift3);
 
+  //# on :: (b -> b -> c) -> (a -> b) -> a -> a -> c
```

Comment:
We must decide whether the type of the first argument should be `b -> b -> c` or `(b, b) -> c`. Currently the type signature and the implementation contradict each other.

There are a handful of higher-order functions for which we provide both forms. In each case `S.<name>` takes a curried function while `S.<name>_` takes an uncurried function. We could provide both `S.on` and `S.on_`, but `S.on` should take a `b -> b -> c` function regardless.